File: csrc/com/eagle/programmar/C/Types/C_FunctionPointer.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Aug 8, 2024

namespace com.eagle.programmar.C.Types
{
	using C_FunctionParameter = com.eagle.programmar.C.C_Function.C_FunctionParameter;
	using C_Function_ParameterDefs = com.eagle.programmar.C.C_Function.C_Function_ParameterDefs;
	using C_MoreParameterDefs = com.eagle.programmar.C.C_Function.C_MoreParameterDefs;
	using C_Program = com.eagle.programmar.C.C_Program;
	using C_Type = com.eagle.programmar.C.C_Type;
	using C_Field_Definition = com.eagle.programmar.C.Symbols.C_Field_Definition;
	using C_Comment = com.eagle.programmar.C.Terminals.C_Comment;
	using C_KeywordChoice = com.eagle.programmar.C.Terminals.C_KeywordChoice;
	using TokenList = com.eagle.tokens.TokenList;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationLeftParen = com.eagle.tokens.punctuation.PunctuationLeftParen;
	using PunctuationRightParen = com.eagle.tokens.punctuation.PunctuationRightParen;
	using PunctuationSemicolon = com.eagle.tokens.punctuation.PunctuationSemicolon;
	using PunctuationStar = com.eagle.tokens.punctuation.PunctuationStar;

	public class C_FunctionPointer : TokenSequence
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) @OPT C_KeywordChoice scope = new com.eagle.programmar.C.Terminals.C_KeywordChoice(com.eagle.programmar.C.C_Program.getModifiers());
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.C.C_Type jtype;
		public C_Type jtype;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.tokens.punctuation.PunctuationLeftParen leftParen1;
		public PunctuationLeftParen leftParen1;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.tokens.punctuation.PunctuationStar star;
		public PunctuationStar star;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) com.eagle.programmar.C.Symbols.C_Field_Definition id;
		public C_Field_Definition id;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) @OPT C_Function_ParameterDefs weirdExtraParameters;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(70) com.eagle.tokens.punctuation.PunctuationRightParen rightParen1;
		public PunctuationRightParen rightParen1;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(80) com.eagle.tokens.punctuation.PunctuationLeftParen leftParen2;
		public PunctuationLeftParen leftParen2;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(90) @OPT C_Comment comment;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(100) @OPT C_FunctionParameter param;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(110) @OPT TokenList<com.eagle.programmar.C.C_Function.C_MoreParameterDefs> moreParams;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(120) com.eagle.tokens.punctuation.PunctuationRightParen rightParen2;
		public PunctuationRightParen rightParen2;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(130) @OPT PunctuationSemicolon semicolon;
		public  OPT;
	}

}

File: src/com/eagle/programmar/C/Types/C_FunctionPointer.java
// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Aug 8, 2024

package com.eagle.programmar.C.Types;

import com.eagle.programmar.C.C_Function.C_FunctionParameter;
import com.eagle.programmar.C.C_Function.C_Function_ParameterDefs;
import com.eagle.programmar.C.C_Function.C_MoreParameterDefs;
import com.eagle.programmar.C.C_Program;
import com.eagle.programmar.C.C_Type;
import com.eagle.programmar.C.Symbols.C_Field_Definition;
import com.eagle.programmar.C.Terminals.C_Comment;
import com.eagle.programmar.C.Terminals.C_KeywordChoice;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationLeftParen;
import com.eagle.tokens.punctuation.PunctuationRightParen;
import com.eagle.tokens.punctuation.PunctuationSemicolon;
import com.eagle.tokens.punctuation.PunctuationStar;

public class C_FunctionPointer extends TokenSequence
{
	public @S(10) @OPT C_KeywordChoice scope = new C_KeywordChoice(C_Program.getModifiers());
	public @S(20) C_Type jtype;
	public @S(30) PunctuationLeftParen leftParen1;
	public @S(40) PunctuationStar star;
	public @S(50) @OPT C_KeywordChoice NULLABLE = new C_KeywordChoice("_Nullable", "_Nonnull");
	public @S(60) @OPT C_Field_Definition id;
	public @S(70) @OPT C_Function_ParameterDefs weirdExtraParameters;
	public @S(80) PunctuationRightParen rightParen1;
	public @S(90) PunctuationLeftParen leftParen2;
	public @S(100) @OPT C_Comment comment;
	public @S(110) @OPT C_FunctionParameter param;
	public @S(120) @OPT TokenList<C_MoreParameterDefs> moreParams;
	public @S(130) PunctuationRightParen rightParen2;
	public @S(140) @OPT PunctuationSemicolon semicolon;
}
